[bug]  npx create-tauri-app get error

### Describe the bug

`npx create-tauri-app` with Vanilla.js works,
but other templates all get an error

### Reproduction

```
npx create-tauri-app

We hope to help you create something special with Tauri!
You will have a choice of one of the UI frameworks supported by the greater web tech community.
This tool should get you quickly started. See our docs at https://tauri.studio/

If you haven't already, please take a moment to setup your system.
You may find the requirements here: https://tauri.studio/en/docs/get-started/setup-windows/

Press any key to continue...
? What is your app name? tauri-app
? What should the window title be? Tauri App
? What UI recipe would you like to add? create-vite (vanilla, vue, react, svelte, preact, lit) (https://vitejs.dev/guide/#scaffolding-your-first-vite-project)
? Add "@tauri-apps/api" npm package? Yes
? Which vite template would you like to use? vue
>> Running initial command(s)
npm ERR! could not determine executable to run

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Local\npm-cache\_logs\2022-02-11T02_22_44_672Z-debug-0.log
Error with command: npx
Error: Error: Command failed with exit code 1: npx  create-vite@latest tauri-app --template vue
    at C:\Users\user\AppData\Local\npm-cache\_npx\b58944764482de26\node_modules\create-tauri-app\dist\index.js:62:15
    at Generator.throw (<anonymous>)
    at rejected (C:\Users\user\AppData\Local\npm-cache\_npx\b58944764482de26\node_modules\create-tauri-app\dist\index.js:39:65)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

logs
```
0 verbose cli [
0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
0 verbose cli   'C:\\Users\\user\\AppData\\Roaming\\nvm\\v17.5.0\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'exec',
0 verbose cli   '--',
0 verbose cli   '',
0 verbose cli   'create-vite@latest',
0 verbose cli   'tauri-app',
0 verbose cli   '--template',
0 verbose cli   'vue'
0 verbose cli ]
1 info using npm@8.4.1
2 info using node@v17.5.0
3 timing npm:load:whichnode Completed in 1ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:C:\Users\user\AppData\Roaming\nvm\v17.5.0\node_modules\npm\npmrc Completed in 1ms
6 timing config:load:builtin Completed in 1ms
7 timing config:load:cli Completed in 2ms
8 timing config:load:env Completed in 2ms
9 timing config:load:file:C:\Users\user\Desktop\test\.npmrc Completed in 1ms
10 timing config:load:project Completed in 5ms
11 timing config:load:file:C:\Users\user\.npmrc Completed in 3ms
12 timing config:load:user Completed in 3ms
13 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in 1ms
14 timing config:load:global Completed in 1ms
15 timing config:load:validate Completed in 2ms
16 timing config:load:credentials Completed in 1ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 21ms
19 timing npm:load:configload Completed in 21ms
20 timing npm:load:setTitle Completed in 1ms
21 timing config:load:flatten Completed in 4ms
22 timing npm:load:display Completed in 5ms
23 verbose logfile C:\Users\user\AppData\Local\npm-cache\_logs\2022-02-11T02_22_44_672Z-debug-0.log
24 timing npm:load:logFile Completed in 10ms
25 timing npm:load:timers Completed in 0ms
26 timing npm:load:configScope Completed in 0ms
27 timing npm:load Completed in 38ms
28 silly logfile start cleaning logs, removing 4 files
29 http fetch GET 200 https://registry.npmjs.org/undefined 27ms (cache stale)
30 timing command:exec Completed in 44ms
31 verbose stack Error: could not determine executable to run
31 verbose stack     at getBinFromManifest (C:\Users\user\AppData\Roaming\nvm\v17.5.0\node_modules\npm\node_modules\libnpmexec\lib\get-bin-from-manifest.js:17:23)
31 verbose stack     at exec (C:\Users\user\AppData\Roaming\nvm\v17.5.0\node_modules\npm\node_modules\libnpmexec\lib\index.js:115:15)
31 verbose stack     at async module.exports (C:\Users\user\AppData\Roaming\nvm\v17.5.0\node_modules\npm\lib\cli.js:66:5)
32 verbose pkgid undefined@0.1.0
33 verbose cwd C:\Users\user\Desktop\test
34 verbose Windows_NT 10.0.19043
35 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\user\\AppData\\Roaming\\nvm\\v17.5.0\\node_modules\\npm\\bin\\npm-cli.js" "exec" "--" "" "create-vite@latest" "tauri-app" "--template" "vue"
36 verbose node v17.5.0
37 verbose npm  v8.4.1
38 error could not determine executable to run
39 verbose exit 1
40 timing npm Completed in 1587ms
41 verbose code 1
42 error A complete log of this run can be found in:
42 error     C:\Users\user\AppData\Local\npm-cache\_logs\2022-02-11T02_22_44_672Z-debug-0.log
```

### Expected behavior

_No response_

### Platform and versions

```shell
windows10
node 17.5
```


### Stack trace

_No response_

### Additional context

_No response_

--- a/.changes/cta-crash.md
+++ b/.changes/cta-crash.md
@@ -1,0 +1,5 @@
+---
+"create-tauri-app": patch
+---
+
+Fix `create-tauri-app` failing to bootstrap projects.

--- a/tooling/create-tauri-app/src/shell.ts
+++ b/tooling/create-tauri-app/src/shell.ts
@@ -24,7 +24,7 @@ export const shell = async (
       if (log) console.log(`[running]: ${command}`)
       return await execa(
         command,
-        args.filter((e) => e != ''),
+        args.filter((e) => e !== ''),
         {
           stdio: 'inherit',
           cwd: process.cwd(),

--- a/tooling/create-tauri-app/src/shell.ts
+++ b/tooling/create-tauri-app/src/shell.ts
@@ -22,12 +22,16 @@ export const shell = async (
       })
     } else {
       if (log) console.log(`[running]: ${command}`)
-      return await execa(command, args, {
-        stdio: 'inherit',
-        cwd: process.cwd(),
-        env: process.env,
-        ...options
-      })
+      return await execa(
+        command,
+        args.filter((e) => e != ''),
+        {
+          stdio: 'inherit',
+          cwd: process.cwd(),
+          env: process.env,
+          ...options
+        }
+      )
     }
   } catch (error) {
     console.error('Error with command: %s', command)

